File: frontend/lib/models/receipt.dart
class ReceiptItem {
  final int id;
  final String productName;
  final double quantity;
  final double unitPrice;
  final double totalPrice;
  final String category;

  ReceiptItem({
    required this.id,
    required this.productName,
    required this.quantity,
    required this.unitPrice,
    required this.totalPrice,
    required this.category,
  });

  factory ReceiptItem.fromJson(Map<String, dynamic> json) {
    return ReceiptItem(
      id: json['id'],
      productName: json['product_name'],
      quantity: (json['quantity'] as num).toDouble(),
      unitPrice: (json['unit_price'] as num).toDouble(),
      totalPrice: (json['total_price'] as num).toDouble(),
      category: json['category'],
    );
  }
}

class Receipt {
  final int id;
  final String storeName;
  final double totalAmount;
  final DateTime createdAt;
  final List<ReceiptItem> items;

  Receipt({
    required this.id,
    required this.storeName,
    required this.totalAmount,
    required this.createdAt,
    required this.items,
  });

  factory Receipt.fromJson(Map<String, dynamic> json) {
    return Receipt(
      id: json['id'],
      storeName: json['store_name'],
      totalAmount: (json['total_amount'] as num).toDouble(),
      createdAt: DateTime.parse(json['created_at']),
      items: (json['items'] as List)
          .map((i) => ReceiptItem.fromJson(i))
          .toList(),
    );
  }
}

File: backend/app/models/receipt.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime, timezone

from app.database import Base


class Receipt(Base):
    __tablename__ = "receipts"

    id = Column(Integer, primary_key=True, index=True)
    store_name = Column(String, default="Unknown")
    total_amount = Column(Float, default=0.0)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    image_path = Column(String, nullable=True)

    items = relationship("ReceiptItem", back_populates="receipt", cascade="all, delete-orphan")


class ReceiptItem(Base):
    __tablename__ = "receipt_items"

    id = Column(Integer, primary_key=True, index=True)
    receipt_id = Column(Integer, ForeignKey("receipts.id"))
    product_name = Column(String)
    quantity = Column(Float, default=1.0)
    unit_price = Column(Float, default=0.0)
    total_price = Column(Float, default=0.0)
    category = Column(String, default="Other")

    receipt = relationship("Receipt", back_populates="items")
